docs/servicedns-with-externaldns.md

```diff
@@ -0,0 +1,250 @@
+$ gcloud dns record-sets transaction add ns-cloud-b{1..4}.googledomains.com. \
+    --name "external.example.com." --ttl 300 --type NS --zone "example-com"
+$ gcloud dns record-sets transaction execute --zone "example-com"
+```
+
+Deploy the ExternalDNS controller. The `domain-filter` argument is used to filter hosted zones matching
+domain name `external.example.com`. Omit this flag to process all available hosted zones.
+```bash
+$ cat <<EOF | kubectl create -f -
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: external-dns
+  namespace: federation-system
+spec:
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: external-dns
+    spec:
+      containers:
+      - name: external-dns
+        image: registry.opensource.zalan.do/teapot/external-dns:v0.5.7
+        args:
+        - --source=crd
+        - --crd-source-apiversion=multiclusterdns.federation.k8s.io/v1alpha1
+        - --crd-source-kind=DNSEndpoint
+        - --provider=google
+        - --domain-filter=external.example.com
+        - --registry=txt
+        - --txt-owner-id=my-identifier
+EOF
+```
+
+Deploy test `FederatedDeployment` and `FederatedService` resources:
+```bash
+$ cat <<EOF | kubectl create -f -
+apiVersion: core.federation.k8s.io/v1alpha1
+kind: FederatedDeployment
+metadata:
+  name: test-deployment
+spec:
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      replicas: 2
+      selector:
+        matchLabels:
+          app: nginx
+      template:
+        metadata:
+          labels:
+            app: nginx
+        spec:
+          containers:
+          - image: nginx
+            name: nginx
+---
+apiVersion: core.federation.k8s.io/v1alpha1
+kind: FederatedDeploymentPlacement
+metadata:
+  name: test-deployment
+spec:
+  clusterNames:
+  - cluster1
+  - cluster2
+---
+apiVersion: core.federation.k8s.io/v1alpha1
+kind: FederatedService
+metadata:
+  name: test-service
+spec:
+  template:
+    spec:
+      selector:
+        app: nginx
+      type: LoadBalancer
+      ports:
+        - name: http
+          port: 80
+---
+apiVersion: core.federation.k8s.io/v1alpha1
+kind: FederatedServicePlacement
+metadata:
+  name: test-service
+spec:
+  clusterNames:
+  - cluster1
+  - cluster2
+EOF
+```
+
+Check the status of all the resources in each cluster by running:
+```bash
+$ for r in deployment service; do
+    for c in cluster1 cluster2; do
+        echo; echo ------------ ${c} resource: ${r} ------------; echo
+        kubectl --context=${c} get ${r}
+        echo; echo
+    done
+done
+```
+
+It may take a few minutes for the `EXTERNAL-IP` field of each `Service` to be populated. Next, create the
+`MultiClusterServiceDNSRecord`:
+```bash
+$ cat <<EOF | kubectl create -f -
+apiVersion: multiclusterdns.federation.k8s.io/v1alpha1
+kind: MultiClusterServiceDNSRecord
+metadata:
+  name: test-service
+spec:
+  federationName: federation-system
```

Comment:
nitpick, federationName `federation-system` is same with the namespace name which federation controller plane deployed, so the dns name `test-service.default.federation-system.svc.external.example.com` is a little confused.
I thought in the begin it's `<service-name>.<namespace>.<namespace>.svc.external.example.com`.

Reply:
`<service-name>.<namespace>.<namespace>.svc.external.example.com.` is not correct. I am updating the doc and will provide the correct naming scheme.